apps/readest-app/src/app/reader/components/annotator/Annotator.tsx

```diff
@@ -584,7 +898,7 @@ const Annotator: React.FC<{ bookKey: string }> = ({ bookKey }) => {
       if (!booknoteGroups[href]) {
         booknoteGroups[href] = { id, href, label, booknotes: [] };
       }
-      booknoteGroups[href].booknotes.push(booknote);
+      booknoteGroups[href]!.booknotes.push(booknote); //booknoteGroups[href] is initialized in above case if originally undefined
```

Comment:
The non-null assertion operator (!) does not change the runtime behavior of the compiled JavaScript. If this triggers a warning in ESLint or your IDE, please verify that you’re using the correct ESLint version and that your IDE is referencing the proper tsconfig settings.